### PR TITLE
feat(mojaloop/project#2573): automate install and basic test across 3 k8s versions

### PR DESCRIPTION
**Pull request by lewisdaly**
This adds automated testing across 3 k8s versions.

I had some trouble getting the backend dependencies up and running: I can get everything working except for `kratos`.

See [this issue](https://app.zenhub.com/workspaces/mojaloop-project-59edee71d1407922110cf083/issues/mojaloop/project/2578) for more information. I disabled `kratos` for now

Followup to this PR:
- add a `helm test` step after the install

### CODE REVIEW

**Comment by lewisdaly** [comment]
@mdebarros @vijayg10 @elnyry-sam-k this is ready for review now. Please let me know what you think!

**Comment by lewisdaly** [comment]
@vijayg10, thanks for the fix, but it looks like you broke something else with that update!

From the kubectl log:
```

107s        Warning   FailedMount               pod/oathkeeper-77f8bd56d6-zh8p4                 MountVolume.SetUp failed for volume "oathkeeper-rules-volume" : configmap "oathkeeper-rules" not found
9m27s       Normal    Sync                      ingress/oathkeeper-proxy                        Scheduled for sync
10m         Normal    Scheduled                 pod/kratos-db-0                                 Successfully assigned default/kratos-db-0 to ip-172-28-8-57
10m         Normal    Pulling                   pod/oathkeeper-maester-6d6f99dd77-fdrlz         Pulling image "oryd/oathkeeper-maester:v0.1.5"
10m         Normal    Pulling                   pod/kratos-569647c8f9-n88fg                     Pulling image "mysql"
10m         Normal    Pulling                   pod/reporting-events-db-556cd8d679-nbvhg        Pulling image "docker.io/bitnami/mongodb:4.4.10-debian-10-r0"
10m         Normal    Pulling                   pod/kafka-exporter-5f944cd476-7pzdz             Pulling image "docker.io/bitnami/kafka-exporter:1.3.1-debian-10-r88"
10m         Normal    SuccessfulCreate          statefulset/keto-db                             create Pod keto-db-0 in StatefulSet keto-db successful
10m         Normal    Scheduled                 pod/keto-7b47b5d4c8-f7dlt                       Successfully assigned default/keto-7b47b5d4c8-f7dlt to ip-172-28-8-57
10m         Normal    SuccessfulCreate          replicaset/keto-7b47b5d4c8                      Created pod: keto-7b47b5d4c8-f7dlt
10m         Normal    SuccessfulCreate          statefulset/be-zookeeper                        create Pod be-zookeeper-0 in StatefulSet be-zookeeper successful
10m         Normal    SuccessfulCreate          statefulset/kratos-db                           create Pod kratos-db-0 in StatefulSet kratos-db successful
10m         Normal    Pulling                   pod/be-zookeeper-0                              Pulling image "docker.io/bitnami/zookeeper:3.7.0-debian-10-r127"
9m59s       Warning   FailedMount               pod/kratos-db-0                                 MountVolume.SetUp failed for volume "kratos-db-token-jx6w4" : failed to sync secret cache: timed out waiting for the condition
9m59s       Normal    Pulling                   pod/keto-db-0                                   Pulling image "docker.io/bitnami/mysql:8.0.26-debian-10-r31"
9m58s       Warning   FailedMount               pod/mysql-0                                     MountVolume.SetUp failed for volume "custom-init-scripts" : failed to sync configmap cache: timed out waiting for the condition
9m58s       Warning   FailedMount               pod/keto-7b47b5d4c8-f7dlt                       MountVolume.SetUp failed for volume "keto-config-volume" : failed to sync configmap cache: timed out waiting for the condition
9m58s       Warning   FailedMount               pod/keto-7b47b5d4c8-f7dlt                       MountVolume.SetUp failed for volume "keto-token-4v7g7" : failed to sync secret cache: timed out waiting for the condition
9m59s       Warning   FailedMount               pod/kratos-db-0                                 MountVolume.SetUp failed for volume "custom-init-scripts" : failed to sync configmap cache: timed out waiting for the condition
9m59s       Warning   FailedMount               pod/mysql-0                                     MountVolume.SetUp failed for volume "config" : failed to sync configmap cache: timed out waiting for the condition
9m59s       Warning   FailedMount               pod/mysql-0                                     MountVolume.SetUp failed for volume "mysql-token-fp272" : failed to sync secret cache: timed out waiting for the condition
9m59s       Warning   FailedMount               pod/kafka-0                                     MountVolume.SetUp failed for volume "jmx-config" : failed to sync configmap cache: timed out waiting for the condition
9m59s       Warning   FailedMount               pod/kratos-db-0                                 MountVolume.SetUp failed for volume "config" : failed to sync configmap cache: timed out waiting for the condition
9m54s       Normal    Pulling                   pod/kratos-db-0                                 Pulling image "docker.io/bitnami/mysql:8.0.26-debian-10-r31"
9m54s       Normal    Pulling                   pod/kafka-0                                     Pulling image "docker.io/bitnami/kafka:2.8.0-debian-10-r84"
9m53s       Normal    Pulling                   pod/keto-7b47b5d4c8-f7dlt                       Pulling image "oryd/keto:v0.6.0-alpha.1-sqlite"
9m53s       Normal    Pulling                   pod/mysql-0                                     Pulling image "docker.io/bitnami/mysql:8.0.26-debian-10-r31"
9m53s       Normal    Pulled                    pod/oathkeeper-maester-6d6f99dd77-fdrlz         Successfully pulled image "oryd/oathkeeper-maester:v0.1.5" in 6.571043805s
9m52s       Normal    Created                   pod/oathkeeper-maester-6d6f99dd77-fdrlz         Created container oathkeeper-maester
9m52s       Normal    Started                   pod/oathkeeper-maester-6d6f99dd77-fdrlz         Started container oathkeeper-maester
8m47s       Normal    Created                   pod/kafka-exporter-5f944cd476-7pzdz             Created container kafka-exporter
8m46s       Normal    Started                   pod/kafka-exporter-5f944cd476-7pzdz             Started container kafka-exporter
9m35s       Normal    Pulled                    pod/kafka-exporter-5f944cd476-7pzdz             Successfully pulled image "docker.io/bitnami/kafka-exporter:1.3.1-debian-10-r88" in 24.757503155s
9m34s       Normal    Pulled                    pod/keto-7b47b5d4c8-f7dlt                       Successfully pulled image "oryd/keto:v0.6.0-alpha.1-sqlite" in 18.90487794s
8m47s       Normal    Pulled                    pod/kafka-exporter-5f944cd476-7pzdz             Container image "docker.io/bitnami/kafka-exporter:1.3.1-debian-10-r88" already present on machine
8m23s       Normal    Created                   pod/keto-7b47b5d4c8-f7dlt                       Created container keto
8m23s       Normal    Started                   pod/keto-7b47b5d4c8-f7dlt                       Started container keto
9m2s        Warning   BackOff                   pod/kafka-exporter-5f944cd476-7pzdz             Back-off restarting failed container
9m15s       Normal    Pulled                    pod/kratos-569647c8f9-n88fg                     Successfully pulled image "mysql" in 45.028905406s
9m15s       Normal    Created                   pod/reporting-events-db-556cd8d679-nbvhg        Created container mongodb
9m15s       Normal    Created                   pod/kratos-569647c8f9-n88fg                     Created container wait-for-mysql
9m15s       Normal    Pulled                    pod/reporting-events-db-556cd8d679-nbvhg        Successfully pulled image "docker.io/bitnami/mongodb:4.4.10-debian-10-r0" in 44.908491175s
9m14s       Normal    Started                   pod/reporting-events-db-556cd8d679-nbvhg        Started container mongodb
9m14s       Normal    Started                   pod/kratos-569647c8f9-n88fg                     Started container wait-for-mysql
9m9s        Normal    Created                   pod/be-zookeeper-0                              Created container zookeeper
9m9s        Normal    Pulled                    pod/be-zookeeper-0                              Successfully pulled image "docker.io/bitnami/zookeeper:3.7.0-debian-10-r127" in 50.506912406s
9m9s        Normal    Created                   pod/kratos-db-0                                 Created container mysql
9m9s        Normal    Pulled                    pod/mysql-0                                     Successfully pulled image "docker.io/bitnami/mysql:8.0.26-debian-10-r31" in 44.645173421s
9m9s        Normal    Pulled                    pod/kratos-db-0                                 Successfully pulled image "docker.io/bitnami/mysql:8.0.26-debian-10-r31" in 44.953209037s
9m9s        Normal    Started                   pod/be-zookeeper-0                              Started container zookeeper
9m8s        Normal    Created                   pod/mysql-0                                     Created container mysql
9m8s        Normal    Created                   pod/keto-db-0                                   Created container mysql
9m8s        Normal    Pulled                    pod/keto-db-0                                   Successfully pulled image "docker.io/bitnami/mysql:8.0.26-debian-10-r31" in 50.544090689s
9m7s        Normal    Started                   pod/keto-db-0                                   Started container mysql
9m7s        Normal    Pulling                   pod/mysql-0                                     Pulling image "docker.io/bitnami/mysqld-exporter:0.13.0-debian-10-r75"
9m7s        Normal    Started                   pod/mysql-0                                     Started container mysql
9m7s        Normal    Started                   pod/kratos-db-0                                 Started container mysql
9m4s        Normal    Pulling                   pod/kafka-0                                     Pulling image "docker.io/bitnami/jmx-exporter:0.16.1-debian-10-r41"
9m4s        Normal    Pulled                    pod/kafka-0                                     Successfully pulled image "docker.io/bitnami/kafka:2.8.0-debian-10-r84" in 49.318466565s
9m5s        Normal    Pulled                    pod/mysql-0                                     Successfully pulled image "docker.io/bitnami/mysqld-exporter:0.13.0-debian-10-r75" in 2.54935753s
9m4s        Normal    Started                   pod/mysql-0                                     Started container metrics
9m4s        Normal    Started                   pod/kafka-0                                     Started container kafka
9m4s        Normal    Created                   pod/mysql-0                                     Created container metrics
9m4s        Normal    Created                   pod/kafka-0                                     Created container kafka
8m23s       Warning   Unhealthy                 pod/keto-7b47b5d4c8-f7dlt                       Liveness probe failed: Get "http://10.42.0.18:4467/health/alive": dial tcp 10.42.0.18:4467: connect: connection refused
8m31s       Warning   Unhealthy                 pod/keto-7b47b5d4c8-f7dlt                       Readiness probe failed: Get "http://10.42.0.18:4467/health/ready": dial tcp 10.42.0.18:4467: connect: connection refused
8m57s       Normal    Pulled                    pod/kafka-0                                     Successfully pulled image "docker.io/bitnami/jmx-exporter:0.16.1-debian-10-r41" in 6.804488066s
8m57s       Normal    Created                   pod/kafka-0                                     Created container jmx-exporter
8m57s       Normal    Started                   pod/kafka-0                                     Started container jmx-exporter
8m53s       Warning   Unhealthy                 pod/kafka-0                                     Liveness probe failed: dial tcp 10.42.0.15:9092: connect: connection refused
8m39s       Warning   Unhealthy                 pod/keto-db-0                                   Startup probe failed: mysqladmin: [Warning] Using a password on the command line interface can be insecure.
mysqladmin: connect to server at 'localhost' failed
error: 'Can't connect to local MySQL server through socket '/opt/bitnami/mysql/tmp/mysql.sock' (2)'
Check that mysqld is running and that the socket: '/opt/bitnami/mysql/tmp/mysql.sock' exists!
8m37s       Warning   Unhealthy                 pod/mysql-0                                     Startup probe failed: mysqladmin: [Warning] Using a password on the command line interface can be insecure.
mysqladmin: connect to server at 'localhost' failed
error: 'Can't connect to local MySQL server through socket '/opt/bitnami/mysql/tmp/mysql.sock' (2)'
Check that mysqld is running and that the socket: '/opt/bitnami/mysql/tmp/mysql.sock' exists!
8m45s       Warning   Unhealthy                 pod/kratos-db-0                                 Startup probe failed: mysqladmin: [Warning] Using a password on the command line interface can be insecure.
mysqladmin: connect to server at 'localhost' failed
error: 'Can't connect to local MySQL server through socket '/opt/bitnami/mysql/tmp/mysql.sock' (2)'
Check that mysqld is running and that the socket: '/opt/bitnami/mysql/tmp/mysql.sock' exists!
8m27s       Normal    Pulling                   pod/kratos-569647c8f9-n88fg                     Pulling image "oryd/kratos:v0.5.5-alpha.1-sqlite"
8m26s       Normal    Created                   pod/kratos-569647c8f9-n88fg                     Created container kratos
8m26s       Normal    Pulled                    pod/kratos-569647c8f9-n88fg                     Successfully pulled image "oryd/kratos:v0.5.5-alpha.1-sqlite" in 1.066726883s
8m26s       Normal    Started                   pod/kratos-569647c8f9-n88fg                     Started container kratos
8m23s       Normal    Pulled                    pod/keto-7b47b5d4c8-f7dlt                       Container image "oryd/keto:v0.6.0-alpha.1-sqlite" already present on machine
8m23s       Normal    Killing                   pod/keto-7b47b5d4c8-f7dlt                       Container keto failed liveness probe, will be restarted
66s         Warning   FailedMount               pod/oathkeeper-77f8bd56d6-zh8p4                 Unable to attach or mount volumes: unmounted volumes=[oathkeeper-rules-volume], unattached volumes=[oathkeeper-config-volume oathkeeper-rules-volume oathkeeper-secrets-volume]: timed out waiting for the condition
5m41s       Warning   FailedMount               pod/oathkeeper-77f8bd56d6-zh8p4                 Unable to attach or mount volumes: unmounted volumes=[oathkeeper-rules-volume], unattached volumes=[oathkeeper-rules-volume oathkeeper-secrets-volume oathkeeper-config-volume]: timed out waiting for the condition
```

I think it's related to `managedAccessRules: false` for oathkeeper

File: scripts/k8s-versions-test.sh [hunk 1]
```diff
@@ -22,36 +22,12 @@ function add_hosts {
...
-                printf  "    Updating chart: $chart    "
-                rm -rf $BASE_DIR/charts/$chart/charts 
-                if [[ `helm dep up $BASE_DIR/charts/$chart --skip-refresh 2>&1 | grep "Error" ` != "" ]] ; then 
-                        printf "[failed]\n"
-                        printf "Error updating chart:  %s \n" $chart 
-                else 
-                        printf "[ok]\n" $?
-                fi
-        done
+        su - $k8s_user -c "helm repo add mojaloop http://mojaloop.io/helm/repo/ > /dev/null 2>&1 "
```
Comment (mdebarros) [inline scripts/k8s-versions-test.sh:25]: I dont think any of these are necessary except for the ingress-nginx.

All other dependencies will be pulled by our helm charts.

Comment (lewisdaly) [reply scripts/k8s-versions-test.sh:25]: Ok, I'll take them out

**Review by mdebarros — APPROVED** [review]
+1

**Pull request merged**